rpc_util.go

```diff
@@ -166,6 +167,15 @@ type afterCall func(c *callInfo)
 func (o afterCall) before(c *callInfo) error { return nil }
 func (o afterCall) after(c *callInfo)        { o(c) }
 
+// CallMetadata returns a CallOption that will attach the given metadata
+// to the call headers.
+func CallMetadata(k, v string) CallOption {
```

Comment:
Hmm, actually, what do you think about: `func CallMetadata(kv ...string) CallOption` (whose `beforeCall` does `c.callMetadata = append(c.callMetadata, kv)`)?

This would have the same signature as `metadata.Pairs`. Like `Pairs`, the function should `panic` if `len(kv) % 2 != 0`.

This would allow users that want to attach several pieces of metadata to reuse their own slice specifying them.

Reply:
I think the interface becomes more complex and suffers for it (e.g., I could see a user asking, if I have two values for one key, do I attach them as `k1, v1, k1, v2` or `k1, v1, v2`).

On the other hand, users of this feature are probably pretty advanced-level and aren't averse to reading the usage comments? I'm happy to implement though, just wanted to share my thought on it. I'll have the change up soon.

Reply:
How about if we called it `CallMetadataPairs`?  That would more closely mirror the `metadata.Pairs` name, and also help to document its behavior.

Reply:
👍  that makes sense to me!